File: MichiMark/Persistence/Repository/Settings/Member/MemberRepository.swift
import Foundation

protocol MemberRepository: Sendable {
    /// 論理削除されていない Action を全件取得
    func fetchAll() async throws -> [MemberDomain]

    /// 新規作成 or 上書き保存
    func save(_ member: MemberDomain) async throws

    /// 既存 Action の更新
    func update(_ member: MemberDomain) async throws
}

File: flutter/lib/repository/member_repository.dart
import '../domain/master/member/member_domain.dart';

/// メンバーの永続化インターフェース
///
/// 削除は論理削除（isDeleted フラグ）で行うため、deleteメソッドは持たない。
/// 新規作成・更新は save で一本化する（upsert）。
abstract interface class MemberRepository {
  /// 論理削除されていないメンバーを全件取得
  Future<List<MemberDomain>> fetchAll();

  /// メンバーを保存（新規作成 / 上書き更新）
  Future<void> save(MemberDomain member);
}
